internal/ingress/types_equals.go

```diff
@@ -259,16 +259,16 @@ func (s1 *Server) Equal(s2 *Server) bool {
 	if s1.SSLPassthrough != s2.SSLPassthrough {
 		return false
 	}
-	if s1.SSLCertificate != s2.SSLCertificate {
+	if s1.SSLCert.PemFileName != s2.SSLCert.PemFileName {
 		return false
 	}
-	if s1.SSLFullChainCertificate != s2.SSLFullChainCertificate {
+	if s1.SSLCert.FullChainPemFileName != s2.SSLCert.FullChainPemFileName {
 		return false
 	}
-	if !s1.SSLExpireTime.Equal(s2.SSLExpireTime) {
+	if !s1.SSLCert.ExpireTime.Equal(s2.SSLCert.ExpireTime) {
 		return false
 	}
-	if s1.SSLPemChecksum != s2.SSLPemChecksum {
+	if s1.SSLCert.PemSHA != s2.SSLCert.PemSHA {
```

Comment:
I'd suggest to implement `Equal` function for `SSLCert` type and just call `s1.SSLCert.Equal(s2.SSLCert)` here - that would be consistent with the code base here.